### ISSUE TITLE
Suggestion: allow simple one-liner methods

### ISSUE THREAD

**Issue by igorsantos07** [squizlabs/PHP_CodeSniffer#2333]
This is based on an issue I'm having on the project I work on.
While I was looking for a solution for it (and after digging through [code](https://github.com/squizlabs/PHP_CodeSniffer/blob/master/src/Standards/Squiz/Sniffs/Functions/MultiLineFunctionDeclarationSniff.php) and not really understanding its relationship with the affected rules), I stumbled upon a [StackOverflow question](https://stackoverflow.com/questions/43275253/phpcs-allow-simple-methods-in-one-line) that summarizes exactly my problem, so I'm quoting it here:

>I find the code
>```php
>$ids = array_map(function ($entity) { return $entity->getId(); }, $entities);
>```
>much more readable than
>```php
>$ids = array_map(function ($entity) { 
>    return $entity->getId(); 
>}, $entities);
>```
>
> However, the PHPCS forces me to use the second style, because the first one fails with the following for PSR-2 styles:
>
>- Squiz.Functions.MultiLineFunctionDeclaration.ContentAfterBrace says that Opening brace must be the last content on the line
>- Squiz.WhiteSpace.ScopeClosingBrace.ContentBefore says that Closing brace must be on a line by itself
>- Generic.Formatting.DisallowMultipleStatements.SameLine says that Each PHP statement must be on a line by itself
>While I generally agree with these checks in other places, in this case they are completely wrong.
>
> Is there a rule that would say to PHPCS that it should allow Simple methods in one line just like PHPStorm does?

**Comment by calguy1000** [comment]
I also would like a sniff for this type of thing for empty methods or very simple methods with a single statement.

i.e:  
```php
protected function __construct() {}
```
or
```
public function someAbstractMethod() ; bool { return TRUE; }
